crates/store/re_log_encoding/src/rrd/frames.rs

```diff
@@ -190,12 +205,178 @@ impl Decodable for StreamHeader {
     }
 }
 
+// ---
+
+/// The closing frame in an RRD stream. Keeps track of where the [`RrdFooter`] can be found.
+///
+/// During normal operations, there can only be a single [`StreamFooter`] per RRD stream.
+/// It is possible to break that invariant by concatenating streams using external tools,
+/// e.g. by doing something like `cat *.rrd > all_my_recordings.rrd`.
+/// Passing that stream back through Rerun tools, e.g. `cat *.rrd | rerun rrd merge > all_my_recordings.rrd`,
+/// would once again guarantee that only one stream footer is present though.
+/// I.e. that invariant holds as long as one stays within our ecosystem of tools.
+///
+/// [`RrdFooter`]: [crate::RrdFooter]
+#[derive(Debug, Clone, Copy, PartialEq, Eq)]
+pub struct StreamFooter {
+    /// Same as the one in the [`StreamHeader`], i.e. [`crate::rrd::RRD_FOURCC`].
+    ///
+    /// Used to go straight to the footer of an RRD stream ("backwards").
+    pub fourcc: [u8; 4], // RRF2
+
+    /// A unique identifier to disambiguate the footer from other frames.
+    ///
+    /// Always set to [`Self::RRD_IDENTIFIER`].
+    pub identifier: [u8; 4], // FOOT
+
+    /// The position in bytes where the serialized [`RrdFooter`] payload starts, excluding the
+    /// message header.
+    ///
+    /// I.e. a transport-level [`RrdFooter`] can be decoded from the bytes at (pseudo-code):
+    /// ```text
+    /// let start = stream_footer.rrd_footer_byte_offset_from_start_excluding_header;
+    /// let end = start + stream_footer.rrd_footer_byte_size_excluding_header;
+    /// let bytes = &file[start..end];
+    /// let rrd_footer = re_protos::RrdFooter::decode(bytes)?;
+    /// let rrd_footer = rrd_footer.to_application()?;
+    /// ```
+    ///
+    /// [`RrdFooter`]: [crate::RrdFooter]
+    pub rrd_footer_byte_offset_from_start_excluding_header: u64,
+
+    /// The size in bytes of the serialized [`RrdFooter`] payload, excluding the message header.
+    ///
+    /// This is guaranteed to be the same value as the `len` found in the associated message
+    /// header, but duplicating it here makes it possible for decoders to get everything they
+    /// need using a single IO.
+    ///
+    /// [`RrdFooter`]: [crate::RrdFooter]
+    pub rrd_footer_byte_size_excluding_header: u64,
```

Comment:
Again, could be using `re_span::Span` just to reduce the number of variables the reader has to have in their head